ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/cache/HttpCacheEntry.kt

```diff
@@ -70,7 +70,13 @@ internal fun HttpResponse.cacheExpires(isShared: Boolean, fallback: () -> GMTDat
 
     val maxAge = cacheControl.firstOrNull { it.value.startsWith(maxAgeKey) }
         ?.value?.split("=")
-        ?.get(1)?.toInt()
+        ?.get(1)?.let { value ->
+            if (value.all { it.isDigit() }) {
+                value.toIntOrNull() ?: Int.MAX_VALUE
```

Comment:
Seems like it would make sense to use Long here instead for one less cast and allow for larger numbers.  Though it's not like anyone would ever notice 😄

Reply:
Yes, thank you! It was my first thought, but I abandoned it by accident :)